mithril-common/src/chain_observer/pallas_observer.rs

```diff
@@ -0,0 +1,245 @@
+use anyhow::anyhow;
+use async_trait::async_trait;
+use pallas_network::facades::NodeClient;
+use pallas_network::miniprotocols::localstate::queries_v16;
+use std::path::{Path, PathBuf};
+use thiserror::Error;
+
+use crate::chain_observer::interface::*;
+use crate::chain_observer::{ChainAddress, TxDatum};
+use crate::crypto_helper::{KESPeriod, OpCert};
+use crate::entities::StakeDistribution;
+use crate::CardanoNetwork;
+use crate::{entities::Epoch, StdResult};
+
+use super::CardanoCliChainObserver;
+
+/// A runner that uses Pallas library to interact with a Cardano node using N2C Ouroboros mini-protocols
+pub struct PallasChainObserver {
+    socket: PathBuf,
+    network: CardanoNetwork,
+    fallback: Option<super::cli_observer::CardanoCliChainObserver>,
```

Comment:
As the fallback is needed until we completely remove it, the `Option` is not required (this will also simplify some code below). Also using an import would help readability:
```suggestion
    fallback: CardanoCliChainObserver,
```

Reply:
done!